assets/app/features/upgrade/controllers/upgrade-landing.controller.spec.js

```diff
@@ -79,6 +211,69 @@ describe('Upgrade Landing Controller', function() {
         should.exist(controller);
     });
 
+    describe('when created while prepare is not running', function() {
+        it('should not start polling for status', function() {
+            expect(upgradeStatusFactory.waitForStepToEnd).not.toHaveBeenCalled();
+        });
+
+        it('prepare should not be running', function() {
+            assert.isFalse(controller.prepare.running);
+        });
+
+        it('prepare should not be completed', function() {
+            assert.isFalse(controller.prepare.completed);
+        });
+    });
+
+    describe('when created while prepare is running', function() {
+        beforeEach(function() {
+            // local change in mocked service
+            upgradeFactory.getStatus =
+                sinon.spy(function () { return $q.when(incompleteUpgradeResponse); });
+
+            controller = $controller('UpgradeLandingController');
+
+            $rootScope.$digest();
+        });
+
+        it('should start polling for status', function() {
+            expect(upgradeStatusFactory.waitForStepToEnd).toHaveBeenCalled();
+        });
+
+        it('prepare should be running', function() {
+            assert.isTrue(controller.prepare.running);
+        });
+
+        it('prepare should not be completed', function() {
+            assert.isFalse(controller.prepare.completed);
+        });
+    });
+
+    describe('when created while prepare is finished', function() {
```

Comment:
Changed it to "when created after prepare is finished". This is to express that we're testing the creation procedure not the behavior of controller when the prepare step is finished.